### PR TITLE
Introduce GitHub Actions PR Check workflow, fixes JBIDE-27695

### PR DESCRIPTION
**Pull request by odockal**
Signed-off-by: Ondrej Dockal <odockal@redhat.com>

### CODE REVIEW

**Comment by odockal** [comment]
@sbouchet review please.

File: .github/workflows/build_test.yml [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+      matrix:
+        os: [ubuntu-latest]
+        java: ["11"]
+      fail-fast: false
+
+    steps:
+    - name: Checkout PR branch
+      uses: actions/checkout@v2
+      with:
+        ref: ${{ github.event.pull_request.head.sha }}
```
Comment (sbouchet) [inline .github/workflows/build_test.yml:21]: please add fetch-depth: 0 to avoid baseline errors

File: .github/workflows/build_test.yml [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+        path: ~/.m2/repository
+        key: jbosstools-${{ runner.os }}-maven-${{ hashFiles('**/pom.xml') }}
+        restore-keys: |
+          jbosstools-${{ runner.os }}-maven-
+
+    # Build and compile using Maven
+    - name: Build/Compile and run unit tests
+      uses: GabrielBB/xvfb-action@v1
+      with:
+        run: mvn clean verify --settings build-ci/maven-settings.xml --quiet
```
Comment (sbouchet) [inline .github/workflows/build_test.yml:51]: --quiet is difficult to review when failing. please use only --no-transfer-progress

**Review by sbouchet — CHANGES_REQUESTED** [review]

**Review by sbouchet — APPROVED** [review]
LGTM

**Pull request merged**